keps/sig-release/3744-stay-on-supported-go-versions/README.md

```diff
@@ -0,0 +1,444 @@
+     * defaulting GODEBUG in kube-apiserver `main()` to preserve SHA-1 x509 
+       enablement in Kubernetes 1.23.x updating from go1.17 to go1.18
+     * allowing kubectl plugin resolution to locate plugin binaries in the 
+       current directory updating from go1.18 to go1.19
+
+The [go update handbook](https://github.com/kubernetes/sig-release/blob/master/release-engineering/handbooks/go.md)
+and [go update issue template](https://github.com/kubernetes/release/blob/master/.github/ISSUE_TEMPLATE/dep-golang.md)
+would be updated or augmented to include this information.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Test Plan
+
+This KEP is not proposing specific code changes or additions to Kubernetes,
+so it relies on the standard presubmit and post-submit release-blocking/release-informing test jobs.
+
+This KEP does depend on the ability to test a specific change against two go versions.
+Currently, this is accomplished using copies of presubmit jobs suffixed with
+["go-canary"](https://grep.app/search?q=go-canary&filter[repo][0]=kubernetes/test-infra),
+built using a proposed go version and manually triggered on a pull request.
+This is the mechanism currently used to verify changes made to the default Kubernetes
+development branch in preparation for a go minor version update.
+
+To ensure release branches continue to remain compatible with the original
+go minor version used with the .0 release for each release branch,
+presubmit/periodic unit and integration test job variants would be created using using the original
+go minor version. The [per-branch test job config fork](https://github.com/kubernetes/test-infra/tree/master/releng/config-forker)
+tool could be updated to set up these unit and integration test jobs automatically.
+
+##### Prerequisite testing updates
+
+n/a
+
+##### Unit tests
+
+Addition of release branch periodic / presubmit unit test jobs
+running with the original go minor version for the branch.
+
+##### Integration tests
+
+Addition of release branch periodic / presubmit integration test jobs
+running with the original go minor version for the branch.
+
+##### e2e tests
+
+n/a
+
+### Graduation Criteria
+
+n/a
+
+### Upgrade / Downgrade Strategy
+
+n/a
+
+### Version Skew Strategy
+
+n/a
+
+## Production Readiness Review Questionnaire
```

Comment:
I think the PRR is basically non-applicable for this KEP, so what is provided here seems fine. 
I also think you could just specify N/A for everything. I'm not sure enablement/rollback is really relevant here, beyond rolling to a new version as this is concerned with the build of Kubernetes itself. 

Will let @johnbelamaric take a final pass for approval.

Reply:
Yeah, end user cannot possibly enable / rollback this one. 

Distros / vendors can override the go used in their custom builds.

I agree this is basically all N/A for PRR, though it's worth noting that Go updates have been touchy for reliability in the past and this KEP is clearly aimed at avoiding those reliability concerns while moving to stay on supported versions, so it's certainly relevant to the production readiness crowd :-)